packages/dduf/src/check-dduf.ts

```diff
@@ -0,0 +1,217 @@
+import { checkFilename } from "./check-filename";
+import { createBlob } from "@huggingface/blob";
+
+export interface DDUFFileEntry {
+	type: "file"; 
+	name: string; size: number; 
+	fileHeaderOffset: number;
+}
+
+export async function* checkDDUF(
+	url: Blob | URL,
+	opts?: { log?: (x: string) => void }
+): AsyncGenerator<{ type: "file"; name: string; size: number; fileHeaderOffset: number }> {
+	const blob = url instanceof Blob ? url : await createBlob(url);
+
+	opts?.log?.("File size: " + blob.size);
+
+	// DDUF is a zip file, uncompressed.
+
+	const last100kB = await blob.slice(blob.size - 100000, blob.size).arrayBuffer();
+
+	const view = new DataView(last100kB);
+
+	let index = view.byteLength - 22;
+	let found = false;
+
+	while (index >= 0) {
+		if (view.getUint32(index, true) === 0x06054b50) {
+			found = true;
+			break;
+		}
+
+		index--;
+	}
+
+	if (!found) {
+		throw new Error("DDUF footer not found in last 100kB of file");
+	}
+
+	opts?.log?.("DDUF footer found at offset " + (blob.size - last100kB.byteLength + index));
+
+	const diskNumber = view.getUint16(index + 4, true);
+
+	opts?.log?.("Disk number: " + diskNumber);
+
+	if (diskNumber !== 0 && diskNumber !== 0xffff) {
+		throw new Error("Multi-disk archives not supported");
+	}
+
+	let fileCount = view.getUint16(index + 10, true);
+	let centralDirSize = view.getUint32(index + 12, true);
+	let centralDirOffset = view.getUint32(index + 16, true);
+	const isZip64 = centralDirOffset === 0xffffffff;
+
+	opts?.log?.("File count: " + fileCount);
+
+	if (isZip64) {
+		opts?.log?.("Zip64 format detected");
+
+		index -= 20;
+		while (index >= 0) {
+			if (view.getUint32(index, true) === 0x07064b50) {
+				found = true;
```

Comment:
does it matter that variable `found` is already true from DDUF footer not found in last 100kB of file